omp/matrix/dense_kernels.cpp

```diff
@@ -384,13 +362,17 @@ void convert_to_hybrid(std::shared_ptr<const OmpExecutor> exec,
         while (col < num_cols) {
             auto val = source->at(row, col);
             if (val != zero<ValueType>()) {
+                size_type current_coo_idx;
+                // Use the critical section for accessing the coo_idx only, the
+                // rest can be performed in parallel since the index is unique
 #pragma omp critical
                 {
-                    coo_val[coo_idx] = val;
-                    coo_col[coo_idx] = col;
-                    coo_row[coo_idx] = row;
-                    coo_idx++;
+                    current_coo_idx = coo_idx;
+                    ++coo_idx;
```

Comment:
`current_coo_idx = coo_idx++;` does the same in one statement. Pick the one you like better

Reply:
True, but they should perform about the same, right?
I mostly chose the current one to make it two distinct statements, so it is clear from the start what it does.
But I would be fine with either version.

Reply:
Sure, the current version is fine.